### PR TITLE
whisperX Integration

### PR DESCRIPTION
**Pull request by hsnfirooz**
This PR integrates [WhisperX](https://github.com/m-bain/whisperX) to our current pipeline. WhisperX enables batch computation which leads to faster inference. The current status is x25 realtime speed.

**Major changes:**
- WhisperX integration
- GPU based sbatch job submission
- Optimize sbatch time duration instead of default 24h
- Optimize `batch_size` based on GPU available memory to decrease inference time and avoiding CUDA OutOfMemory error.
- Using `ffmpeg` in a subprocess for faster input file conversion
- Parallelizing `Pynnoate` diarization and 'WhisperX` transcription for 2x pipeline speed

**Required Files**
WhisperX requires [this VAD model](https://whisperx.s3.eu-west-2.amazonaws.com/model_weights/segmentation/0b5b3216d60a2d32fc086b47ea8c67589aaeb26b7e07fcbe620d6d0b83e209ea/pytorch_model.bin) to be available in `TORCH_HOME` directory. `TORCH_HOME` can be accessed via `torch.hub._get_torch_home()` and in `speech2text` module is located at `/scratch/shareddata/speech2text`

**Environment Variables**
For `SPEECH2TEXT_CPUS_PER_TASK`, 6 is enough as we are using

### CODE REVIEW

File: src/submit.py [hunk 1]
```diff
@@ -121,6 +120,57 @@ def create_array_input_file(input_dir, output_dir, job_name, tmp_dir):
...
+    total_duration = "00:00:00"
+    total_loading = "00:00:00"
+
+    if Path(input_path).suffix == ".json":
+        with open(input_path, "r") as fin:
+            input_files = json.load(fin)
+    else:
+        input_files=input_path
+        
+    for audio_file in input_files:
```
Comment (teemu-ruokolainen) [inline src/submit.py:152]: This bugs me a bit: if `input_path` does not end in ".json" then is it a string? In that case, would `for audio_file in input_files` loop over the characters in that string?

**Review by teemu-ruokolainen — CHANGES_REQUESTED** [review]
Looks good!

One comment in submit.py:estimate_job_time and then I would run black + isort before merging.

File: src/submit.py (hunk 1 above)
Comment (hsnfirooz) [reply src/submit.py:153]: Thanks for the heads up.
`input_path` is not a string if the input is file, but it's `PosixPath`. However `PosixPath` is not iterable and it raises an error.
Fix the bug  [532177c](https://github.com/AaltoRSE/speech2text/pull/3/commits/532177c0a8cd843495fb574cf70307b3745dfc8e).

**Comment by teemu-ruokolainen** [comment]
Looks good to me! I ran black + isort and I think you can squash+merge this.

**Pull request merged**